src/Cake.Common/Tools/NuGet/Pack/NuGetPacker.cs

```diff
@@ -175,6 +176,18 @@ private ProcessArgumentBuilder GetArguments(FilePath filePath, NuGetPackSettings
                 builder.Append(settings.Verbosity.Value.ToString().ToLowerInvariant());
             }
 
+            // Properties
+            if (settings.Properties != null && settings.Properties.Count > 0)
+            {
+                if (settings.Properties.Values.Any(x => x == null))
```

Comment:
Good call. That's certainly a valid dictionary from an object standpoint, but since all the whitespace will just get converted to spaces in the command line statement that the NuGet argument parser will probably see as empty, it's probably invalid input.

I'll adjust and add some more supporting tests.